chargebee/src/main/java/com/chargebee/android/restore/CBRestorePurchaseManager.kt

```diff
@@ -0,0 +1,144 @@
+package com.chargebee.android.restore
+
+import android.util.Log
+import com.chargebee.android.Chargebee
+import com.chargebee.android.ErrorDetail
+import com.chargebee.android.billingservice.CBPurchase
+import com.chargebee.android.billingservice.GPErrorCode
+import com.chargebee.android.billingservice.RestorePurchaseCallback
+import com.chargebee.android.exceptions.CBException
+import com.chargebee.android.exceptions.ChargebeeResult
+import com.chargebee.android.loggers.CBLogger
+import com.chargebee.android.models.*
+import com.chargebee.android.models.ResultHandler
+import com.chargebee.android.resources.RestorePurchaseResource
+
+
+class CBRestorePurchaseManager {
+
+    companion object {
+        private var allTransactions = ArrayList<PurchaseTransaction>()
+        private var restorePurchases = ArrayList<CBRestoreSubscription>()
+        private var activeTransactions = ArrayList<PurchaseTransaction>()
+        lateinit var completionCallback: RestorePurchaseCallback
+
+        private fun retrieveStoreSubscription(
+            purchaseToken: String,
+            completion: (ChargebeeResult<Any>) -> Unit
+        ) {
+            try {
+                val logger = CBLogger(name = "restore", action = "retrieve_restore_subscriptions")
+                ResultHandler.safeExecuter(
+                    { RestorePurchaseResource().retrieveStoreSubscription(purchaseToken) },
+                    completion,
+                    logger
+                )
+            } catch (exp: Exception) {
+                ChargebeeResult.Error(
+                    exp = CBException(
+                        error = ErrorDetail(
+                            exp.message
+                        )
+                    )
+                )
+            }
```

Comment:
Minor: Can we format this better?

Reply:
try-catch block not required here so removed it. because the `safeExecuter` method uses try catch and handling all the error and success cases throwing callbacks.